Introduce helper for handling of post-handshake handshake messages in TLS <=1.2

hanno-arm opened the pull request:
Handling the receipt of a handshake record after the initial handshake requires non-trivial logic depending on the protocol version and the endpoint. This logic is currently embedded in `mbedtls_ssl_read()`.

With the introduction of support for [D]TLS 1.3, the logic will become even more complex, since [D]TLS 1.3 drops support for renegotiation -- which in [D]TLS 1.2 is the main purpose of post-handshake handshake messages -- but instead introduces numerous other post-handshake handshake messages.

In order to pave the way for those changes, this PR improves readability and maintainability of `mbedtls_ssl_read()` by moving the TLS <=1.2 logic for handling post-handshake handshake messages into a separate helper function 
 `ssl_handle_hs_message_post_handshake()`.

The logic of the code should be entirely unchanged.

gilles-peskine-arm commented on library/ssl_msg.c:
```diff
@@ -5527,135 +5652,15 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
+                return( ret );
             }
 
-            /* At this point, we don't know whether the renegotiation has been
```
Why did you remove this comment? It still seems useful to me.

hanno-arm replied:
Ok, I'm re-introducing it.

gilles-peskine-arm commented on library/ssl_msg.c:
```diff
@@ -5429,6 +5429,131 @@ static int ssl_check_ctr_renegotiate( mbedtls_ssl_context *ssl )
+ */
+static int ssl_handle_hs_message_post_handshake( mbedtls_ssl_context *ssl )
+{
+    int ret;
```
```suggestion
    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
```

hanno-arm replied:
Fixing.

gilles-peskine-arm reviewed (changes requested):
I replayed part of the change locally then looked at the remaining diff with this PR. Looks good to me apart from two minor issues.

paul-elliott-arm reviewed (commented):
Generally agreed with Gilles. Looks like a good refactor, other than that.

hanno-arm commented:
@gilles-peskine-arm @paul-elliott-arm Thank you for your review, and apologies for the long silence. I've addressed your comments. Could you please re-review?

hanno-arm commented:
@gilles-peskine-arm @paul-elliott-arm I rebased to resolve a merge conflict stemming from the removal of SSLv3 renegotiation rejection which was part of the code being moved into `ssl_handle_hs_message_post_handshake()`.

Could you please re-review?

paul-elliott-arm reviewed (approved):
LGTM
